CMakeLists.txt

```diff
@@ -1657,6 +1657,12 @@ elseif(UNIX AND NOT APPLE AND NOT RISCOS AND NOT HAIKU)
     set(HAVE_RPATH TRUE)
   endif()
 
+  if(QNX)
+    # QNX's *printf() family generates a SIGSEGV if NULL is passed for a string
+    # specifier (on purpose), but SDL expects "(null)". Use the built-in
+    # implementation.
+    set (HAVE_VSNPRINTF 0)
```

Comment:
You might also create a custom QNX wrapper that checks for NULL and then dispatches to the system vsnprintf(). You could either hardcode QNX for that, or add a define that indicates that the system vsnprintf() doesn't handle NULL, like NEED_VSNPRINTF_NULL_WRAPPER or something.

Using the built-in snprintf is fine, but it isn't as fully featured as most system implementations (doesn't handle %e, etc.)

Reply:
To do that I'll need to parse the format string to see which argument is a string. I may end up succumbing to peer pressure and change the QNX `*printf()` implementation, though I have some strong feelings about treating NULL specially...

I think that for now using the SDL `vsnprintf` is fine. We'll see if there are any issues.